src/raster/processing/dem/aspect.rs

```diff
@@ -0,0 +1,125 @@
+use super::options::common_dem_options;
+use crate::cpl::CslStringList;
+use crate::raster::processing::dem::DemSlopeAlg;
+use std::num::NonZeroUsize;
+
+/// Configuration options for [`aspect()`][super::aspect()].
+#[derive(Debug, Clone, Default)]
+pub struct AspectOptions {
+    input_band: Option<NonZeroUsize>,
+    compute_edges: bool,
+    output_format: Option<String>,
+    additional_options: CslStringList,
+    algorithm: Option<DemSlopeAlg>,
+    zero_for_flat: Option<bool>,
+    trigonometric: Option<bool>,
+}
+
+impl AspectOptions {
+    /// Create a DEM-aspect options set.
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    common_dem_options!();
+
+    /// Specify the slope computation algorithm.
+    pub fn with_algorithm(&mut self, algorithm: DemSlopeAlg) -> &mut Self {
+        self.algorithm = Some(algorithm);
+        self
+    }
+
+    /// Return `0` for flat areas with `slope=0`, instead of `-9999`.
+    ///
+    /// See: [`zero_for_flat`](https://gdal.org/programs/gdaldem.html#cmdoption-zero_for_flat)
+    pub fn with_zero_for_flat(&mut self, state: bool) -> &mut Self {
+        self.zero_for_flat = Some(state);
+        self
+    }
+
+    /// Return trigonometric angle instead of azimuth. Thus 0° means East, 90° North, 180° West, 270° South.
+    pub fn with_trigonometric_angles(&mut self, state: bool) -> &mut Self {
+        self.trigonometric = Some(state);
+        self
+    }
+
+    /// Render relevant common options into [`CslStringList`] values, as compatible with
+    /// [`gdal_sys::GDALDEMProcessing`].
+    pub fn to_options_list(&self) -> CslStringList {
```

Comment:
Same as for `store_common_options_to`, but more user-facing we could return `Result<CslStringList>` here. These are unlikely to fail but might, for example on OOM.

Reply:
https://github.com/metasim/gdal/pull/1